frontend/src/framework/internal/components/SelectEnsemblesDialog/selectEnsemblesDialog.tsx

```diff
@@ -155,68 +159,68 @@ export const SelectEnsemblesDialog: React.FC<SelectEnsemblesDialogProps> = (prop
         return false;
     }
 
-    function hasAnyRegularEnsembleChanged(): boolean {
-        return !isEqual(props.selectedRegularEnsembles, newlySelectedRegularEnsembles);
-    }
-
-    function handleAddRegularEnsemble(newItem: RegularEnsembleItem) {
-        if (newlySelectedRegularEnsembles.some((i) => isSameEnsembleItem(i, newItem))) return;
+    function handleAddRegularEnsemble(newItem: InternalRegularEnsembleSetting) {
+        if (selectedRegularEnsembles.some((el) => el.ensembleIdent.equals(newItem.ensembleIdent))) {
+            return;
+        }
 
-        setNewlySelectedRegularEnsembles((prev) => [...prev, newItem]);
+        setSelectedRegularEnsembles((prev) => [...prev, newItem]);
     }
 
-    function handleUpdateRegularEnsemble(updatedItem: RegularEnsembleItem) {
-        setNewlySelectedRegularEnsembles((prev) => {
-            return prev.map((item) => (isSameEnsembleItem(item, updatedItem) ? updatedItem : item));
+    function handleUpdateRegularEnsemble(updatedItem: InternalRegularEnsembleSetting) {
+        setSelectedRegularEnsembles((prev) => {
+            return prev.map((el) => (el.ensembleIdent.equals(updatedItem) ? updatedItem : el));
```

Comment:
I missed this in previous review:

`return prev.map((el) => (el.ensembleIdent.equals(updatedItem) ? updatedItem : el));` -> `return prev.map((el) => (el.ensembleIdent.equals(updatedItem.ensembleIdent) ? updatedItem : el));`